src/eventStore/index.ts

```diff
@@ -1,15 +1,20 @@
-import { EventStoreConfig } from './interface';
+import { EventStoreDynamoDB } from './dynamodb';
+import { EventStoreConfig, EventStoreType } from './interface';
 import { EventStoreMongoDB } from './mongodb';
 
-export function createEventStoreInstance(config: EventStoreConfig) {
+export function createEventStoreInstance(
+  config: EventStoreConfig[EventStoreType],
+) {
   const types = {
     mongodb: EventStoreMongoDB,
+    dynamodb: EventStoreDynamoDB,
   };
 
   const EventStoreClass = types[config.type];
   if (!EventStoreClass) {
-    throw new Error("invalid event store type.");
+    throw new Error('invalid event store type.');
   }
 
+  // @ts-ignore
```

Comment:
Maybe above type completion will resolve type error.

Reply:
At this point, the parameter type of the EventStoreClass constructor is never .